commonv5/ts/helpers/request.ts

```diff
@@ -1,46 +1,56 @@
-import axios from "axios";
 import * as semver from "semver";
 import * as tl from "azure-pipelines-task-lib/task";
+import fetch from 'node-fetch'
 import Endpoint from "../sonarqube/Endpoint";
 
 interface RequestData {
   [x: string]: any;
 }
 
+const parseJson = async response => {
+  const text = await response.text()
+  try{
+    const json = JSON.parse(text)
+    return json
+  } catch(err) {
+    return text
+  }
+}
+
 export function get(endpoint: Endpoint, path: string, query?: RequestData): Promise<any> {
   tl.debug(`[SQ] API GET: '${path}' with query "${JSON.stringify(query)}"`);
+  let url = endpoint.url + path;
+  const headers = {}
+
+  if (query) {
+    const params = new URLSearchParams(query)
+    url = url + '?' + params
+  }
+
+  if (endpoint.auth) {
+    headers["Authorization"] = `Basic: ${btoa(endpoint.auth.user + ':' + endpoint.auth.pass)}`
```

Comment:
For the case where `pass` is undefined, I think this would create the string `user:undefined`